package.json

```diff
@@ -58,38 +61,36 @@
   },
   "dependencies": {
     "@conda-store/conda-store-ui": "2023.10.2",
-    "@jupyterlab/application": "^3.1.0",
-    "@jupyterlab/apputils": "^3.1.0",
-    "@jupyterlab/mainmenu": "^3.1.0",
-    "@jupyterlab/settingregistry": "^3.1.0",
-    "@jupyterlab/ui-components": "^3.1.0",
-    "@lumino/widgets": "^1.37.1",
-    "react": "^17.0.1",
-    "react-dom": "^17.0.1"
+    "@jupyterlab/application": "^4.0.9",
+    "@jupyterlab/apputils": "^4.1.2",
+    "@jupyterlab/mainmenu": "^4.0.9",
+    "@jupyterlab/settingregistry": "^4.0.9",
+    "@jupyterlab/ui-components": "^4.0.9",
+    "@lumino/widgets": "^2.3.1"
   },
   "devDependencies": {
     "@babel/core": "^7.0.0",
     "@babel/preset-env": "^7.0.0",
     "@jupyterlab/builder": "^3.0.0 || ^4.0.0",
     "@jupyterlab/testutils": "^3.0.0 || ^4.0.0",
     "@types/jest": "^26.0.0",
-    "@types/react": "^17.0.0",
+    "@types/react": "^18.0.26",
     "@types/react-dom": "^17.0.0",
-    "@typescript-eslint/eslint-plugin": "^4.8.1",
-    "@typescript-eslint/parser": "^4.8.1",
-    "eslint": "^7.14.0",
-    "eslint-config-prettier": "^6.15.0",
-    "eslint-plugin-prettier": "^3.1.4",
-    "jest": "^26.4.2",
+    "@typescript-eslint/eslint-plugin": "~6.13.2",
+    "@typescript-eslint/parser": "~6.13.2",
+    "eslint": "~8.55.0",
+    "eslint-config-prettier": "~9.1.0",
+    "eslint-plugin-prettier": "~5.0.1",
```

Comment:
I see that you are pinning all linting tools. The upside is no new warnings when a new minor version gets released, the downside is no fixes as new minor versions gets released - I think either approach is fine, just documenting it with this comment, as it is easy to glance over the change to `~` without noticing it :)